app/lib/expression_viz_service.rb

```diff
@@ -445,4 +441,61 @@ def self.calculate_median(genes, cell)
     values = genes.map {|gene| gene['scores'][cell].to_f}
     Gene.array_median(values)
   end
+
+  # return a text file for morpheus to use when rendering dotplots/heatmaps
+  # supports both expression data (gct format) and annotation data
+  def self.get_morpheus_text_data(study: nil,
+                                  file_type: nil,
+                                  genes: nil,
+                                  cluster: nil,
+                                  collapse_by: nil,
+                                  selected_annotation: nil)
+    cells = cluster.concatenate_data_arrays('text', 'cells')
+    row_data = []
+    case file_type
+    when :gct
+      headers = %w(Name Description)
+      cols = cells.size
+      cells.each do |cell|
+        headers << cell
+      end
+      rows = []
+      genes.each do |gene|
+        row = [gene['name'], ""]
+        case collapse_by
+        when 'z-score'
```

Comment:
I know this is just a copy-paste, but since we don't expose these in the UI, do we want to leave them?  Or should we add them as collapse metrics since we have code for it?

Reply:
Let's leave them in this for now, since eventually this might get used by a UX feature that does expose them